protostar/starknet/cheatable_cached_state.py

```diff
@@ -174,6 +177,32 @@ def unregister_expected_call(
             if not data_for_address:
                 del self.expected_contract_calls[contract_address]
 
+    def register_expected_call_cairo(
+        self, contract_address: Address, function_name: str, calldata: CairoData
+    ):
+        if self.expected_contract_calls_cairo.get(contract_address):
```

Comment:
What do you mean? I don't see any weird behavior here but maybe I'm missing something?

`self.expected_contract_calls_cairo.get(contract_address)` returns just `None` and leaves the `dict` unchanged (for `defaultdict` as well).

Reply:
Ok, I misunderstood what you meant by saying "consider" here. You're right, but I discovered that to make it work, we have to implement a condition in the state class' `_apply` method. We have to do this:
```
        if parent.expected_contract_calls or self.expected_contract_calls:  # <--- adding this if
            parent.expected_contract_calls = {
                **parent.expected_contract_calls,
                **self.expected_contract_calls,
            }
```
Otherwise, we're going to have `{}` instead of a `defaultdict` object, and eventually, a `KeyError` will be raised here. So either way we have to perform a check.

I think it is better to do it here, it's just safer. If for some reason, `expected_contract_calls` is modified somehow and we get a `{}` here, we'll get a `KeyError`. This simple check here makes sure it never happens. What you proposed is a nice trick but I think it's not worth implementing in this case considering the risk.